lua/entities/gmod_wire_expression2/core/cl_http_cl.lua

```diff
@@ -0,0 +1,131 @@
+--[[
+
+	Clientside HTTP library
+	 - swadical 🍌
+
+]]
+
+--	Clientside HTTP is separate from serverside HTTP:
+--		Serverside HTTP allows greater speeds than clientside HTTP and it is
+--		up to the server owner to decide which extension should be enabled
+--		through a convar (found in core/http.lua)
+
+--	This library will mirror the functionality of core/http.lua as best as
+--	possible, making sure any existing e2s which make use of the http library
+--	are not broken by this extension
+
+E2Lib.clHTTP = {}
+local lib = E2Lib.clHTTP
+
+lib.netMsgID = "wire_expression2_cl_http" -- open to change
+lib.currentUID = 0
+
+-- number function(void)
+function lib:newUID()
+	self.currentUID = (self.currentUID + 1) % 255
+	return self.currentUID -- return a number between 0 and 254
+end
+
+lib.rawRequest = http.Fetch
+lib.segmentSize = 4 * 1024-- 4 kilobytes
+lib.segmentBits = math.min(math.ceil(math.log(lib.segmentSize,2)) + 2,32)
+lib.sendInterval = 0.5-- seconds
+lib.HTTP_REQUEST_FAILED = 1024--enum for HTTP failure
+
+-- void function(string id, number interval, function func)
+-- Internal: Do not call.
+function lib.timerCoroutine(id,interval,func)
+	func = coroutine.wrap(func)
+	timer.Create(id,interval,0,function()
+		if func() then
+			timer.Remove(id)
+		end
+	end)
+end
+
+-- string, string function(void)
+-- Internal: Do not call.
+function lib:decodeRequest()
+	return net.ReadString(),net.ReadUInt(8)
+end
+
+-- void function(void)
+-- Internal: Do not call.
+function lib:handleIncomingRequest()
```

Comment:
I don't like the style of using methods (ie. `obj:func()`) for library functions. Under the hood, calls just turns into `lib.handleIncomingRequest(lib)`, so it's passing an extra argument for no reason.